tests/functional/pv/pvc_snapshot/test_restore_snapshot_when_parent_pvc_deleted.py

```diff
@@ -0,0 +1,228 @@
+                else "fs"
+            )
+            pod_obj.run_io(
+                storage_type=storage_type,
+                size="1G",
+                runtime=20,
+                fio_filename=file_name,
+                end_fsync=1,
+            )
+            log.info(f"IO started on pod {pod_obj.name}")
+        log.info("Started IO on all pods")
+
+        # Wait for IO to finish
+        log.info("Wait for IO to finish on pods")
+        for pod_obj in self.pods:
+            pod_obj.get_fio_results()
+            log.info(f"IO finished on pod {pod_obj.name}")
+            # Calculate md5sum to compare after restoring
+            file_name_pod = (
+                file_name
+                if (pod_obj.pvc.volume_mode == constants.VOLUME_MODE_FILESYSTEM)
+                else pod_obj.get_storage_path(storage_type="block")
+            )
+            pod_obj.pvc.md5sum = pod.cal_md5sum(
+                pod_obj,
+                file_name_pod,
+                pod_obj.pvc.volume_mode == constants.VOLUME_MODE_BLOCK,
+            )
+        log.info("IO finished on all pods")
+
+        # Create snapshots
+        log.info("Creating snapshot of the PVCs")
+        snap_objs = []
+        for pvc_obj in self.pvcs:
+            log.info(f"Creating snapshot of PVC {pvc_obj.name}")
+            snap_obj = snapshot_factory(pvc_obj, wait=False)
+            snap_obj.md5sum = pvc_obj.md5sum
+            snap_obj.interface = pvc_obj.interface
+            snap_objs.append(snap_obj)
+            log.info(f"Created snapshot of PVC {pvc_obj.name}")
+
+        log.info("Snapshots are created. Wait for the snapshots to be in Ready state")
+        for snap_obj in snap_objs:
+            snap_obj.ocp.wait_for_resource(
+                condition="true",
+                resource_name=snap_obj.name,
+                column=constants.STATUS_READYTOUSE,
+                timeout=180,
+            )
+            snap_obj.reload()
+        log.info("Snapshots are Ready")
+
+        # Delete the parent PVCs
+        log.info("Deleting the Parent PVCs and it's attached pod")
+        for pod_obj in self.pods:
+            pod_obj.delete()
+            pod_obj.ocp.wait_for_delete(resource_name=pod_obj.name)
+        for pvc_obj in self.pvcs:
+            log.info(f"Deleting PVC {pvc_obj.name}")
+            pvc_obj.delete()
+            pvc_obj.ocp.wait_for_delete(resource_name=pvc_obj.name)
+
+        # Restore snapshots with different access mode
+        log.info("Restoring snapshots to create new PVCs")
+        restore_pvcs = []
+        for snap_obj in snap_objs:
+            access_modes = access_modes_dict[snap_obj.interface][
+                snap_obj.parent_volume_mode
+            ]
+            for access_mode in access_modes:
+                restore_obj = snapshot_restore_factory(
+                    snapshot_obj=snap_obj,
+                    volume_mode=snap_obj.parent_volume_mode,
+                    access_mode=access_mode,
```

Comment:
how are you making sure that we are not restoring the snapshot with the same access mode? To me, it looks like with the current approach, there will be a few restores with same access mode.